xmppserver/src/main/java/org/jivesoftware/openfire/muc/spi/LocalMUCRoom.java

```diff
@@ -2331,28 +2328,30 @@ public Presence kickOccupant(JID jid, JID actorJID, String actorNickname, String
      */
     private void kickPresence(Presence kickPresence, JID actorJID, String nick) {
         // Get the role(s) to kick
-        List<MUCRole> occupants = new ArrayList<>(occupantsByNickname.get(kickPresence.getFrom().getResource().toLowerCase()));
-        for (MUCRole kickedRole : occupants) {
-            // Add the actor's JID that kicked this user from the room
-            if (actorJID != null && actorJID.toString().length() > 0) {
-                Element frag = kickPresence.getChildElement(
-                        "x", "http://jabber.org/protocol/muc#user");
-                Element actor = frag.element("item").addElement("actor");
-                actor.addAttribute("jid", actorJID.toBareJID());
-                if (nick != null) {
-                    actor.addAttribute("nick", nick);
+        List<MUCRole> occupants = occupantsByNickname.get(kickPresence.getFrom().getResource().toLowerCase());
+        if (occupants != null) {
+            for (MUCRole kickedRole : new ArrayList<>(occupants)) {
```

Comment:
Line 2333 creates a copy of an ArrayList. At this point, an error may occur if the list is changed at the same time, e.g. if the list is changed in the constructor during copying.

I would change the type of the list to a CopyOnWriteList so that this case cannot occur. Is there anything against it?

Reply:
That's a good plan

Reply:
done